js/components/icon.js

```diff
@@ -0,0 +1,28 @@
+var React = require('react');
+
+
+var Icon = React.createClass({
+  shouldComponentUpdate: function(nextProps, nextState) {
+    return  (nextProps.iconClass != this.props.iconClass ||
+            nextProps.iconString != this.props.iconString ||
+            JSON.stringify(nextProps.iconStyle) != JSON.stringify(this.props.iconStyle))
+  },
+
+  render: function() {
+    return (
+      <span className={this.props.iconClass}
+        style={this.props.iconStyle}
+        onMouseOver={this.props.onMouseOver} onMouseOut={this.props.onMouseOut}
+        onClick={this.props.onClick}>
+        {this.props.iconString}
+      </span>
+    );
+  },
+  proptypes: {
+    iconClass: React.PropTypes.string,
```

Comment:
propTypes

consider ordering like this jsx style guide - https://github.com/airbnb/javascript/tree/master/react#ordering